### ISSUE TITLE
There should be an optional argument for limiting parallel spawned processes

### ISSUE THREAD

**Issue by justinwoo** [purescript/psc-package#126]
Probably using http://hackage.haskell.org/package/async-pool-0.9.0.2

Apparently, on Windows on small computers, the mapConcurrently-run spawned git processes end up with timeouts or something.

**Comment by eviefp** [comment]
Still investigating this, will update once I confirm the issue.

**Comment by eviefp** [comment]
I can confirm that changing the `forConcurrently_` with a `for_` and the `mapConcurrently` with `traverse` fixed the issue for us.

Any hints on how you want this solved? How about a `--single-thread` flag?

**Comment by kritzcreek** [comment]
with the RTS options you've already made it only spawn one thread. The single threaded runtime will still spawn multiple processes in green-threads (kind of like node does).

You really don't want to force it to only clone one repo at a time, because that takes wayyy longer. Find a good number that Appveyor can handle and limit the number of active jobs to that.

EDIT:

A quick google turned up this possible solution:

```
import Control.Concurrent.MSem
import Control.Concurrent.Async
import qualified Data.Traversable as T

mapPool :: T.Traversable t => Int -> (a -> IO b) -> t a -> IO (t b)
mapPool max f xs = do
    sem <- new max
    mapConcurrently (with sem . f) xs
```

https://stackoverflow.com/questions/18896103/can-haskells-control-concurrent-async-mapconcurrently-have-a-limit

**Comment by eviefp** [comment]
Okay, then how about `--limit-threads=n`?

I'll test it on our repo first to see if that works.

**Comment by justinwoo** [comment]
Fixed in #127
